src/charm.py

```diff
@@ -52,6 +98,56 @@ def _install_upf_snap(self) -> None:
             logger.error("An exception occurred when installing the UPF snap. Reason: %s", str(e))
             raise e
 
+    def _generate_upf_config_file(self) -> None:
+        """Generate the UPF configuration file."""
+        core_ip_address = self._get_core_network_ip_config()
+        content = render_upf_config_file(
+            upf_hostname=self._get_upf_hostname(),
+            upf_mode=self._get_upf_mode(),
+            access_interface_name=UPF_ACCESS_INTERFACE_NAME,
+            core_interface_name=UPF_CORE_INTERFACE_NAME,
```

Comment:
As these will be running on real hardware, should they not be parameterized?

Reply:
I added config values for interface names